server/src/main/java/org/elasticsearch/transport/InboundDecoder.java

```diff
@@ -160,6 +169,14 @@ private static int headerBytesToRead(BytesReference reference) {
             return fixedHeaderSize;
         } else {
             int variableHeaderSize = reference.getInt(TcpHeader.VARIABLE_HEADER_SIZE_POSITION);
+            if (variableHeaderSize < 0) {
+                throw new StreamCorruptedException("invalid negative variable header size: " + variableHeaderSize);
+            }
+            if (variableHeaderSize > maxVariableHeaderSize.getBytes() - fixedHeaderSize) {
```

Comment:
Good job on handling overflow! 
Nit: should the variable be named `maxHeaderSize` instead of `maxVariableHeaderSize`?

Reply:
Pushed [4878ab4](https://github.com/elastic/elasticsearch/pull/98692/commits/4878ab452e73dd45db0b61fe476737a08bca0ce7)